### PR TITLE
Implement the drop RPC.

### PR DESCRIPTION
**Pull request by aboodman**
Toward https://github.com/rocicorp/repc/issues/159

### CODE REVIEW

**Review by arv — APPROVED** [review]

File: src/kv/idbstore.rs [hunk 1]
```diff
@@ -116,6 +117,23 @@ impl IdbStore {
...
 
+    pub async fn drop_store(name: &str) -> Result<()> {
+        let window = web_sys::window().ok_or(StoreError::Str(str!("could not get window")))?;
+        let factory = window
+            .indexed_db()?
+            .ok_or(StoreError::Str(str!("could not get indexeddb")))?;
+        let request = factory.delete_database(name)?;
+        let (callback, receiver) = IdbStore::oneshot_callback();
+        request.set_onsuccess(Some(callback.as_ref().unchecked_ref()));
+        request.set_onerror(Some(callback.as_ref().unchecked_ref()));
```
[the remark this answers is not in the document]
Comment (aboodman) [reply src/kv/idbstore.rs:129]: Yeah this is a little sketchy in that it relies on idb only calling one of the two callbacks, but it's what we do elsewhere in this file, and seems to work for now.

File: src/embed/dispatch.rs [hunk 2]
```diff
@@ -131,6 +132,19 @@ async fn do_close(conns: &mut ConnMap, req: &Request) -> Response {
     Ok("".into())
 }
 
+async fn do_drop(_: &mut ConnMap, req: &Request) -> Response {
+    match &req.db_name[..] {
+        #[cfg(not(target_arch = "wasm32"))]
+        "mem" => Ok("".into()),
```
Comment (phritz) [inline src/embed/dispatch.rs:138]: curious why just "mem"? is there some guarantee that there is exactly one db and it is named "mem" when in not-browser mode? think i might have expected _ => Ok for not wasm32 and the _ you have below for wasm32.

**Review by phritz — COMMENTED** [review]
LGTM

File: src/embed/dispatch.rs (hunk 2 above)
Comment (aboodman) [reply src/embed/dispatch.rs:138]: The way that you trigger getting access to the mem db is by using the special name "mem". And that is only available in non-wasm (which is a hacky way of saying "test" mode).

**Pull request merged**